pytools/py_codegen.py

```diff
@@ -71,7 +78,9 @@ def __init__(self, name, args, decorators=None):
 
     @property
     def _gen_filename(self):
-        return f"generated_code_for_{self.name}"
+        # Note that the '<ipython-input-' prefix is for compatibility with
+        # line_profiler: https://github.com/pyutils/line_profiler/blob/1630e7c9a295ace2feb1d2b188e68f4d2833fb20/line_profiler/line_profiler.py#L194-L210
+        return f"<ipython-input- generated code for '{self.name}'>"
```

Comment:
Please only use the work-around if `line_profiler` is in `sys.modules`.

Reply:
Done in [68ed96a](https://github.com/inducer/pytools/pull/295/commits/68ed96a881444eb48d7fc207a89c0ab5029cc7a7)